src/storage/query/QueryBaseProcessor.inl

```diff
@@ -522,9 +597,16 @@ kvstore::ResultCode QueryBaseProcessor<REQ, RESP>::collectEdgeProps(
                     continue;
                 }
             }
+
+            // Check if the schema has TTL
+            if (!checkDataExpiredForTTL(schema.get(), reader.get())) {
+                proc(reader.get(), key, props);
```

Comment:
If the ttl exprired,  we could "continue" the loop.

Reply:
Good!